File: MongoClientLibrary/Models/GridWindow.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MongoClientLibrary.Models;

public class GridWindow : Window
{
    public GridWindow(string title, List<string> columnHeaders, List<string> rowHeaders) : base(title)
    {
        
        ColumnHeaders = columnHeaders;
        RowHeaders = rowHeaders;
    }
    public List<string> ColumnHeaders { get; }
    public List<string> RowHeaders { get; }
}

File: NoSqlDemo/ClientApp/src/app/workspace/models/gridWindow.ts
import {WindowModel} from './window.model';

export class GridWindow implements WindowModel {
  constructor(id: string,
              position: [number, number],
              title: string,
              columnHeaders: string[],
              rowHeaders: string[]) {
    this.id = id;
    this.position = position;
    this.title = title;
    this.columnHeaders = columnHeaders;
    this.rowHeaders = rowHeaders;
  }

  id: string;
  position: [number, number];
  title: string;
  columnHeaders: string[];
  rowHeaders: string[];
}
